pkg/utils/authcheck/authcheck.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package authcheck provides utilities to check authentication configuration for data plane resources.
+// File authcheck contains functions to run customized checks inside of a Pod.
+package authcheck
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"golang.org/x/oauth2/google"
+)
+
+const (
+	// resource is used as the path to get the default token from metadata server.
+	// In workload-identity-gsa mode, this path will return a token if
+	// corresponding k8s service account and google service account establish a correct relationship.
+	resource = "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token"
+	// scope is used as the scope to get token from default credential.
+	scope = "https://www.googleapis.com/auth/cloud-platform"
+	// authMessage is the key words to determine if a termination log is about authentication.
+	authMessage = "checking authentication"
+)
+
+type AuthenticationCheck interface {
+	Check(ctx context.Context) error
+}
+
+type DefaultAuthenticationCheck struct {
+	authType AuthType
+	client   *http.Client
+	url      string
+}
+
+func NewDefaultAuthenticationCheck(authType AuthType) AuthenticationCheck {
```

Comment:
A simpler name would be `NewDefault` because the package name already indicates it's for auth check.